toolkit/docs/building/dedicated_guides/build_packages.md

```diff
@@ -0,0 +1,165 @@
+
+## **4. Add/Modify a Package**
+
+See [Working with Packages](https://github.com/microsoft/CBL-MarinerTutorials/blob/-/docs/packages/working_with_packages.md#tutorial-customize-your-image-with-unsupported-packages) from the [CBL-MarinerTutorials](https://github.com/microsoft/CBL-MarinerTutorials) repo for details on working with RPM packages. This will cover:
+
+* Working with source files
+* Calculating package signatures
+* Writing .spec files
+
+The core repo behaves slightly differently than the dedicated build environment described in [CBL-MarinerTutorials](https://github.com/microsoft/CBL-MarinerTutorials), so use the steps described below to build your package instead.
+
+## **5. Build the Package**
+
+### Setup
+
+```bash
+# Clone the CBL-Mariner repo
+git clone https://github.com/microsoft/CBL-Mariner.git
+cd CBL-Mariner
+
+# Sync to the latest stable build
+git checkout 2.0-stable
+cd ./toolkit
+sudo make clean QUICK_REBUILD=y
+```
+
+### Build a toolchain if using an unstable branch
+
+```bash
+# OPTIONAL:
+#   Build the latest toolchain **if you are NOT working on a stable branch**
+git checkout main
+sudo make toolchain QUICK_REBUILD_TOOLCHAIN=y
+
+# Copy it somewhere safe if  you want (you can also just leave QUICK_REBUILD_TOOLCHAIN=y set)
+cp ../build/toolchain/toolchain_built_rpms_all.tar.gz ~/mariner_toolchain.tar.gz
+```
+
+### Decide if you filter out all other packages
+
+There are two ways to build packages:
+
+1) Let the scheduler figure out all of the required packages and build everything from source with the versions in the current branch.
+2) Only build the very specific packages you care about, and get **EVERYTHING ELSE** from the stable releases.
+
+Option #2 is generally faster, but can lead to accuracy issues. If one of your packages' dependencies has also been changed in your branch but not published, the build will use the older published version. This is often fine for quick development work, but is not good enough for final testing.
+
+### Signatures
+
+See [Source Hashes](/toolkit/docs/building/building.md#source-hashes) and [Create a Signature Meta-data File](https://github.com/microsoft/CBL-MarinerTutorials/blob/-/docs/packages/working_with_packages.md#create-a-signature-meta-data-file) for details on how source signatures work. You will need to update the signature files for a package if you change the sources at all.
+
+### Dev Loop
+
+```bash
+# These are the packages we want to iterate on
+package_list="openssh nano"
+
+# OPTIONAL:
+#   If you want a fast (but possibly less accurate) build, only pack the specific packages we want to build and
+#   use upstream stable packages to fulfill all dependencies.
+pkg_filter="SRPM_PACK_LIST='$packages'"
+#   Otherwise, trust the scheduler to optimize the build
+pkg_filter=""
+
+# Modify your package
+touch ../SPECS/nano/nano.spec
+touch ../SPECS/nano/opessh.spec
```

Comment:
would this be `../SPECS/openssh/openssh.spec`

Reply:
Clearly, I'm just adding a new ssh protocol specialized for use in the Midwest (fixed).